app.js

```diff
@@ -219,28 +222,58 @@ app.post('/interactions', verifyKeyMiddleware(process.env.PUBLIC_KEY), async (re
             };
 
             const lbargs = args[0].split('+');
-            if (lbargs[1] ==='totd') {
-                track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
-            }
-            if (lbargs[lbargs.length - 1] === 'f') {
-                args.push(0);
-            } else if (lbargs[lbargs.length - 1] === 'l') {
-                args.push(1000-args[3]);
-            } else if (lbargs[lbargs.length - 1] === 'p') {
-                data.values[0].split(';').forEach((e) => {
-                    args.push(e);
-                });
-            }
+            if (lbargs[1] === 'totd') track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
+            let body;
+            if (lbargs[lbargs.length - 1] ==='i') {
+                let leaderboard = await trackmaniaFacade.getLeaderboardInfo(track_info, args[3], true, args[4]);
+                const userId = message.interaction.user.id;
+                const userName = message.interaction.user.global_name;
+                if (accountWatchers[userId] !== undefined) {
+                    const mapType = lbargs[lbargs.length-3];
+                    const watchedRecords = await trackmaniaFacade.getWatchedAccounts(accountWatchers[userId], revertUID(lbargs[lbargs.length-2], mapType));
+                    console.log(watchedRecords);
+                    const watchingLB = await trackmaniaFacade.generateLeaderboardField({
+                        name: `${userName}'s Watched Players`,
+                        records: watchedRecords,
+                    });
+                    leaderboard.fields.push(watchingLB.field);
+                    leaderboard.watchedAccounts = watchingLB.players;
+                }
+                body = trackmania.embedLeaderboardInfo(leaderboard);
+            } else {
+                console.log(lbargs);
+                if (lbargs[lbargs.length - 1] === 'f') {
+                    args.push(0);
+                } else if (lbargs[lbargs.length - 1] === 'l') {
+                    args.push(1000-args[3]);
+                } else if (lbargs[lbargs.length - 1] === 'p') {
+                    data.values[0].split(';').forEach(e => args.push(e));
+                }
 
-            const lb_info = await trackmaniaFacade.getLeaderboardInfo(track_info, args[3], true, args[4]);
+                console.log(args);
 
-            log.info(args);
+                const { records, buttons, pageSelecter } = await trackmaniaFacade.updateLeaderboard({
+                    groupUid: track_info.groupUid,
+                    mapUid: track_info.mapUid,
+                }, Number(args[3]), true, Number(args[4]));
```

Comment:
nit: The `onlyWorld` boolean param in the API contract for [Get map leaderboard](https://webservices.openplanet.dev/live/leaderboards/top) is kinda poo and it has infested our code. When I look at this line, I see the `true` and don't really have much context. In another PR, I wonder if we could add a constant and use it in our calls to these facade methods:
```js
...
// somewhere in global constants, probably in trackmania.js
const LeaderboardType = {
    ONLY_WORLD: "lb_type_only_world",
    ALL: "lb_type_all"
};
...
// in the argument parsing logic for our API call
let onlyWorld;
if (leaderboardType == LeaderboardType.ONLY_WORLD) {
    onlyWorld = true;
} else if (leaderboardType == LeaderboardType.ALL) {
    onlyWorld = false;
} else {
    throw new Error(`Invalid leaderboard type "${leaderboardType}"`);
}
// do API call...
...
// function definition
    async updateLeaderboard(lbInfo, leaderboardType, length = 25, offset = 0) {
...
// call I've highlighted
const leaderboardInfo = {
    groupUid: track_info.groupUid,
    mapUid:  track_info.mapUid,
};
const { records, buttons, pageSelecter } = await trackmaniaFacade.updateLeaderboard(leaderboardInfo,
    LeaderboardType.ONLY_WORLD, length, offset);
```

Reply:
onlyWorld be like
![true](https://github.com/user-attachments/assets/22c8d86d-d769-412d-9695-3ee2e2864af1)